wasm/main.go

```diff
@@ -0,0 +1,62 @@
+//go:build js && wasm
+
+package main
+
+import (
+	"crypto/ed25519"
+	"encoding/base64"
+	"encoding/json"
+	"syscall/js"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+)
+
+/*
+ * This is the glue to bind the functions into javascript so they can be called
+ */
+func main() {
+	done := make(chan struct{})
+
+	// Bind the functions to javascript
+	js.Global().Set("sayHello", js.FuncOf(sayHello))
+	js.Global().Set("generateKey", js.FuncOf(generateKey))
+	js.Global().Set("makeDid", js.FuncOf(makeDid))
+
+	<-done
+}
+
+// 1. Simplest function - note we wrap things with js.ValueOf (if a primitive you don't technically need to)
+func sayHello(_ js.Value, args []js.Value) interface{} {
+	return js.ValueOf("Hello from golang via wasm!")
+}
+
+// 2. Calling a ssi-sdk function directly - but returning a plain old string
+func generateKey(_ js.Value, args []js.Value) interface{} {
+
+	keyType := args[0].String()
+	for _, k := range crypto.GetSupportedKeyTypes() {
+		if string(k) == keyType {
+			publicKey, _, _ := crypto.GenerateKeyByKeyType(k)
+			pubKeyBytes, _ := crypto.PubKeyToBytes(publicKey)
+			return js.ValueOf(base64.StdEncoding.EncodeToString(pubKeyBytes))
+		}
+	}
+
+	return js.ValueOf("Unknown key type")
+}
+
+// 3. Returning a richer object, converting to json and then unmarshalling to make it a js object
+func makeDid(_ js.Value, args []js.Value) interface{} {
+
+	pubKey, _, _ := crypto.GenerateKeyByKeyType(crypto.Ed25519)
```

Comment:
should we pull out the errs here and below?

Reply:
Another thing to think about with WASM is how it interfaces with storage. It's not quite clear cut. I.e you generate the public key in WASM, but you'll now how to figure out a way to transfer and serialize the key to the application handling the storage. That transfer always felt a little "iffy" to me. I.e you've now sent your private key to your application, and that may be a security risk.

Reply:
@andorsk I am not that worried about that at the moment: I like the idea of using WASM for pure computation (ie functions without side effects and no state outside its bounds) were possible especially with ssi-ssd (people can correct me if I am wrong). I know you could build your whole app and state in wasm, but I don't think that is needed here is it?

Reply:
@michaelneale fair points. I get using WASM for pure computation, and think a stateless app is probably fair.  One question though here: moving keys back and forth between local storage to the WASM application. I don't have an answer here, but does shuffling the key around between WASM and JS runtime render any security issues?

i.e 

```js
var pk = JSON.parse(CreatePK()) // generates PK. Now key is JS Object. You could store this somewhere after.
doX(JSON.stringify(pk)) // PK is serialized and sent. 
```